src/main/java/org/opentripplanner/graph_builder/issue/report/HTMLWriter.java

```diff
@@ -0,0 +1,123 @@
+package org.opentripplanner.graph_builder.issue.report;
+
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import org.opentripplanner.datastore.api.CompositeDataSource;
+import org.opentripplanner.datastore.api.DataSource;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class HTMLWriter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HTMLWriter.class);
+  private final DataSource target;
+  private final Collection<DataImportIssue> issues;
+  private final BucketKey bucketKey;
+  private boolean addGeoJSONLink;
+  private final List<BucketKey> keys;
+
+  HTMLWriter(
+    CompositeDataSource reportDirectory,
+    Bucket bucket,
+    List<BucketKey> keys,
+    boolean addGeoJSONLink
+  ) {
+    this.bucketKey = bucket.key();
+    this.addGeoJSONLink = addGeoJSONLink;
+    LOG.debug("Creating file: {}", bucketKey.key());
+    this.target = reportDirectory.entry(bucketKey.key() + ".html");
+    this.keys = keys;
+    this.issues = bucket.issues();
+  }
+
+  HTMLWriter(CompositeDataSource reportDirectory, String filename, List<BucketKey> keys) {
+    LOG.debug("Creating index file: {}", filename);
+    this.target = reportDirectory.entry(filename + ".html");
+    this.keys = keys;
+    this.issues = null;
+    this.bucketKey = new BucketKey(filename, null);
+  }
+
+  void writeFile() {
+    try (PrintWriter out = new PrintWriter(target.asOutputStream(), true, StandardCharsets.UTF_8)) {
+      printPrelude(out);
+
+      String title = bucketKey.label();
+      out.println(String.format("<h1>OpenTripPlanner data import issue log %s</h1>", title));
+      out.println("<h2>Graph report for <em>graph.obj</em></h2>");
+
+      printCategoryLinks(out);
+
+      if (issues != null) {
+        if (addGeoJSONLink) {
+          out.printf("<a href=\"./%s.geojson\">Open issues in a GeoJSON file</a>", bucketKey.key());
+        }
+
+        writeIssues(out);
+      }
+
+      printPostamble(out);
+    }
+  }
+
+  private void printPrelude(PrintWriter out) {
+    out.println("<html><head><title>Graph report for OTP Graph</title>");
```

Comment:
I think you want to add a `<!DOCTYPE html>` at the beginning so you don't end up in quirks mode.